### ISSUE TITLE
non_ascii_literal false positive: non-ascii in comment in test

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#7739]
Surprisingly the following triggers `non_ascii_literal`.

```rust
#[test]
fn non_ascii_comment() {
    // Ryū crate: https://github.com/dtolnay/ryu
}
```

```console
$ cargo clippy --tests -- -Dclippy::all -Dclippy::pedantic 

error: literal non-ASCII character detected
 --> src/main.rs:2:1
  |
1 |   #[test]
  |   ------- in this procedural macro expansion
2 | / fn non_ascii_comment() {
3 | |     // Ryū crate: https://github.com/dtolnay/ryu
4 | | }
  | |_^
  |
  = note: `-D clippy::non-ascii-literal` implied by `-D clippy::pedantic`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_ascii_literal
  = note: this error originates in the attribute macro `test` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This is unexpected because there are no literals in this code and non-ascii in comments should be totally fine.

<br>

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.57.0-nightly (11491938f 2021-09-29)
binary: rustc
commit-hash: 11491938f80988c7261a1179cf71a25c379c8783
commit-date: 2021-09-29
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```
